### PR TITLE
[checkpoint] add kwargs for load_state_dict

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

**Review by feifeibear — APPROVED** [review]

File: colossalai/utils/checkpoint/module_checkpoint.py [hunk 1]
```diff
@@ -71,22 +71,23 @@ def save_checkpoint(path: str,
...
 
-def load_checkpoint(path,
+def load_checkpoint(path: str,
                     epoch: int,
                     model: torch.nn.Module,
                     optimizer: Optional[ColossalaiOptimizer] = None,
                     lr_scheduler: torch.optim.lr_scheduler._LRScheduler = None,
+                    load_state_dict_kwargs: Optional[Dict] = None,
                     *args,
                     **kwargs):
```
Comment (FrankLeeeee) [inline colossalai/utils/checkpoint/module_checkpoint.py:81]: It can be quite confusing to the user if there are two kwargs

Comment (1SAA) [reply colossalai/utils/checkpoint/module_checkpoint.py:81]: Yeah, there are two functions requiring kwargs.
I think kwargs is conveninent for users.
Should we remove kwargs and add two dict for paramters passing in?

Comment (FrankLeeeee) [reply colossalai/utils/checkpoint/module_checkpoint.py:81]: I think it will be better.

Comment (1SAA) [reply colossalai/utils/checkpoint/module_checkpoint.py]: Done.

**Pull request merged**